### ISSUE TITLE
npm publish fails when run with `yarn run`

### ISSUE THREAD

**Issue by thejameskyle** [lerna/lerna#688]
Not sure what could be causing this

```sh
yarn lerna publish
```
```
npm ERR! need auth auth required for publishing
npm ERR! need auth You need to authorize this machine using `npm adduser`
```

**Comment by evocateur** [comment]
Internally, we are still calling `npm publish --tag <tag>`, even when using `yarn` to drive it (or as `--npm-client`.

However, since v2.0.0-beta.38, there have been [changes to `NpmUtilities.publishTaggedInDir()`](https://github.com/lerna/lerna/compare/v2.0.0-beta.38...master#diff-5463599b8d45d087bae97f0d5df54fdaR115) that addressed an issue with missing `process.env` when the `--registry` flag was provided. The latest code also uses `options.env.cwd` passed to `execSync()` instead of an inline `cd <dir> && npm publish` execution.

Does `lerna@canary` (`2.0.0-beta.38-18-g5043ad2`) fix the issue?

**Comment by jbovenschen** [comment]
For me this still happens at the -2.0.0-rc.4. Any idea what is causing this?

Or is there anything I can do to resolve this issue?

**Comment by shilman** [comment]
Still seeing this issue with `lerna@2.2.0` in the Storybook repo. Our publish process is [documented here](https://github.com/storybooks/storybook/blame/master/CONTRIBUTING.md#L241), and we'd love to use `yarn` across the board, but for now we need to revert to `npm` for this step.

**Comment by jarzuaga** [comment]
I'm on `lerna@2.8.0` and still see this issue. Any news on getting a fix in?

**Comment by evocateur** [comment]
This issue was resolved by #1145 and #1204, which is available since lerna [v2.7.1](https://github.com/lerna/lerna/releases/tag/v2.7.1).

@jarzuaga Please open a new issue with logging output and other details, thanks.
